src/shims/env.rs

```diff
@@ -116,7 +116,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
         let target_os = &this.tcx.sess.target.os;
         assert!(
-            target_os == "linux" || target_os == "macos",
+            matches!(target_os.as_str(), "linux" | "macos" | "android"),
```

Comment:
I feel like we should add a helper `target_os_is_unix()` for this somewhere. Not sure where the appropriate place for such an addition would be though.

Reply:
Well, we have `helpers.rs`. :shrug:

Reply:
However, maybe it would make more sense to check if target_family is unix?

Reply:
Great callout. Added.